src/AvroConvert/BuildSchema/AvroDataContractResolver.cs

```diff
@@ -170,8 +188,8 @@ public override MemberSerializationInfo[] ResolveMembers(Type type)
                 MemberInfo = m.Member,
                 Nullable = m.Nullable,
                 Aliases = _usePropertyNameAsAlias ? new List<string> { m.Member.Name } : new List<string>(),
-                HasDefaultValue = m.DefaultValue != null,
-                DefaultValue = m.DefaultValue
+                HasDefaultValue = m.HasDefaultValue,
```

Comment:
Because the default value may literally be null when specifically decided to be.  ( DefaultValue(null) )  This is important for the Kafka world when deserializing messages of an earlier version of messages  through this version of the schema.